Update Terraform code for COOL multi-account environment

jsf9k opened the pull request:
## 🗣 Description

In this pull request I update this repo to work with our current implementation of the COOL multi-account environment.

## 💭 Motivation and Context

This code is to be deployed to the COOL.  Therefore it must work with our COOL multi-account environment.

## 🧪 Testing

I deployed these changes to production, and I verified that all the pre-commit hooks passed.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Change repo name

jsf9k committed:
Modify backend and providers fro COOL multi-account

jsf9k committed:
Add *.tfvars to .gitignore

jsf9k committed:
Add policy for provisioning FreeIPA

jsf9k committed:
Add new variables to README

jsf9k committed:
Rename a provider

jsf9k committed:
Rename a few outputs

jsf9k committed:
Merge remote-tracking branch 'origin/develop' into improvement/update-for-cool-multiaccount

dav3r commented on provisionfreeipa_policy.tf:
```diff
@@ -0,0 +1,23 @@
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "provisionfreeipa_policy_doc" {
+  # This policy needs to be more restricted, obviously
```
🎗Reminder to create an issue to go back and restrict this policy (if this PR is merged and this policy is still wide open).

dav3r reviewed (approved):
LGTM! 🎉 
See policy reminder in my comment.

felddy commented on .gitignore:
```diff
@@ -1,5 +1,6 @@
 __pycache__
 .python-version
 .terraform
+*.tfvars
```
Yazz!  We might want to make this part of the skeleton.

felddy reviewed (approved):
Looking good PenguinTamer!  Call the ball! ⚽️

jsf9k committed:
Specify the go version

This is required to get around the fact that terraform-docs does not
install with the default version of go that is present.

jsf9k committed:
Go ahead and start using v2 of the setup-go action, even though it's only in beta right now

jsf9k committed:
Merge pull request #14 from cisagov/bugfix/fix-broken-terraform-docs-install

Fix broken terraform-docs install

jsf9k committed:
Merge remote-tracking branch 'skeleton-tf-module/develop' into improvement/update-for-cool-multiaccount

jsf9k committed:
Output resources instead of ARNs and IDs

jsf9k committed:
Specify the COOL Images account as the AMI owner

jsf9k committed:
Adjust for the fact that modules now return resources, not ARNs and IDs

jsf9k committed:
Change default value for cert bucket name

jsf9k committed:
Add remote state for cisagov/cool-sharedaccounts-networking

This allow us to get rid of several input variables.

jsf9k committed:
Specify the string type for variables

Previously string was the default type, but this is no longer the
case:
https://www.terraform.io/docs/configuration/variables.html#type-constraints

I ran into a case today where this mattered.  I used
set([var.string_value]) in a for_each, and Terraform rejected the
syntax since it couldn't verify that the output would be a set of
strings.  Adding the string type declaration to the variable fixed the
issue.
